ingest/src/main/java/com/microsoft/azure/kusto/ingest/resources/RankedStorageAccount.java

```diff
@@ -1,18 +1,93 @@
 package com.microsoft.azure.kusto.ingest.resources;
 
+import com.microsoft.azure.kusto.ingest.utils.TimeProvider;
+
+import java.util.ArrayDeque;
+
 public class RankedStorageAccount {
+    static class Bucket {
+        public int successCount;
+        public int failureCount;
+
+        public Bucket() {
+            this.successCount = 0;
+            this.failureCount = 0;
+        }
+    }
+
+    private final ArrayDeque<Bucket> buckets = new ArrayDeque<>();
     private final String accountName;
+    private final int maxNumberOfBuckets;
+    private final int bucketDurationInSec;
+    private final TimeProvider timeProvider;
+
+    private long lastActionTimestamp;
 
-    public RankedStorageAccount(String accountName) {
+    public RankedStorageAccount(String accountName, int maxNumberOfBuckets, int bucketDurationInSec, TimeProvider timeProvider) {
         this.accountName = accountName;
+        this.maxNumberOfBuckets = maxNumberOfBuckets;
+        this.bucketDurationInSec = bucketDurationInSec;
+        this.timeProvider = timeProvider;
+        lastActionTimestamp = timeProvider.currentTimeMillis();
     }
 
     public void addResult(boolean success) {
```

Comment:
can be simplified and also makes more sense per getRank() calculations:
`

        Bucket lastBucket = adjustForTimePassed();

        lastBucket.total++;

        if (success) {

            lastBucket.successCount++;

        } 
`

Reply:
Great